### PR TITLE
Refactor fetch errors handling

### PR DESCRIPTION
**Pull request by Ndpnt**
- Unify which errors are considered as an `InaccessibleContentError` wether it is the `FullDomFetcher` or `HtmlOnlyFtecher`used. 
- Consider insecure connection as an `InaccessibleContentError`.
- ⚠️ Breaking change: Upgrade Node.js to v14 as it is required by `node-fetch` v3.0.0.

### CODE REVIEW

**Review by martinratinaud — APPROVED** [review]
You may also want to remove the `--insecure-http-parser` in the npm run start and scheduler scripts

I added this to be able to crawl also on insecure connection.

File: src/app/fetcher/fullDomFetcher.js [hunk 1]
```diff
@@ -3,12 +3,11 @@ import puppeteer from 'puppeteer-extra';
...
 
-import logger from '../../logger/index.js';
 import { InaccessibleContentError } from '../errors.js';
 
 puppeteer.use(StealthPlugin());
 
-const PUPPETEER_TIMEOUT = 30 * 1000; // 30s
+const PUPPETEER_TIMEOUT = 30 * 1000; // 30 seconds in ms
 const MAX_RETRIES = 3;
 let browser;
```
Comment (Ndpnt) [inline src/app/fetcher/fullDomFetcher.js:13]: remove `retry` param

Comment (martinratinaud) [reply src/app/fetcher/fullDomFetcher.js:13]: I'm sorry I do not understand @Ndpnt

Comment (Ndpnt) [reply src/app/fetcher/fullDomFetcher.js:13]: No problem, it was a comment for me :)

**Pull request merged**